lib/backend/compat/compat.go

```diff
@@ -251,6 +324,34 @@ func (c *ModelAdaptor) listNodes(l model.NodeListOptions) ([]*model.KVPair, erro
 	return results, nil
 }
 
+// listBlock returns list of KVPairs for Block, includes making sure
+// backwards compatiblity. See getBlock for more details.
+func (c *ModelAdaptor) listBlock(l model.BlockListOptions) ([]*model.KVPair, error) {
+
+	// Get a list of block KVPairs.
+	bal, err := c.client.List(l)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an empty slice of KVPair.
+	results := make([]*model.KVPair, len(bal))
+
+	// Go through the list to make sure Affinity field has a proper value,
+	// and map the value to Affinity if deprecated HostAffinity field is used
+	// by calling getBlock, and populate the KVPair slice to return.
+	for idx, bakv := range bal {
+		kvp, err := c.getBlock(bakv.Key)
```

Comment:
Not a big fan of doing a `List()`, and then doing individual `Get()`s for each item in the list.  This is a lot of round trips to the datastore.

We should just have a `ensureBlockAffinity` type utility which can be applied to the items returned by the above `List()` so we don't need to do so many gets.